### ISSUE TITLE
[Accessibility] Button triggering modal missing aria-haspopup

### ISSUE THREAD

**Issue by amberhinds** [woocommerce/woocommerce-gateway-amazon-pay#331]
## Audit Information
This issue came from an audit WooCommerce had Equalize Digital complete in December 2025. I'm moving the issue out of Linear (WooA11Y-426) to this repo.

- EQD Audit ID: WooExt-518
- Component/Page: Amazon pay modal
- Affected Element: Details about Keep Me Signed In
- URL where issue was discovered: https://apay-woo.mystagingwebsite.com/cart/

## WCAG Compliance

- Success Criterion: 4.1.2 Name, Role, Value
- WCAG Version: 2
- Level: A
- Severity: Medium

## Issue Description

When a button triggers a modal or dialog to open, this should be communicated to screen reader users to set expectations for the action. The aria-haspopup attribute serves this purpose, but it is missing in this case.

### Impacted Populations

Blind and low vision people, Screen reader users (blind/low vision and neurodivergent people)

## Recommended Fix

Add aria-haspopup="dialog" to the button so screen reader users are informed of the action when the button is activated. Additionally, include the aria-controls attribute to link the button to the ID of the modal being opened.

See https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-haspopup  and https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-controls

## Code Example

```
<a id="remember_me_learn_more_link" aria-label="Details about Keep Me Signed In" href="javascript:void(0)" role="button" class="a-popover-trigger a-declarative">
               Details
            <i class="a-icon a-icon-popover"></i></a>
```
## Screenshots

https://drive.google.com/file/d/1olHhq_XPpwmH7zhmi0t3r4FB-dfdWy8G/view?usp=drivesdk

**Comment by ChristianAF** [comment]
Amazon Pay sites Issue
